cardano-api/src/Cardano/Api/Modes.hs

```diff
@@ -171,6 +193,14 @@ data ConsensusModeParams mode where
        :: Byron.EpochSlots
        -> ConsensusModeParams CardanoMode
 
+deriving instance Show (ConsensusModeParams mode)
+
+instance TestEquality ConsensusModeParams where
+  testEquality (ByronModeParams _) (ByronModeParams _) = Just Refl
+  testEquality ShelleyModeParams ShelleyModeParams = Just Refl
+  testEquality (CardanoModeParams _) (CardanoModeParams _) = Just Refl
+  testEquality _ _ = Nothing
```

Comment:
So we need an `Eq` instance (and therefore a `TestEquality` instance) for `ConsensusModeParam`s because we use it in the cli (the command types need an `Eq` instance) to make a `LocalNodeConnectInfo mode` for the queries.

Reply:
Success :smile: